enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,550 @@
+                opacity=1,
+            ),
+            name="Crop Suitability",
+            checked=True,
+        ),
+    ]
+
+
+@APP.callback(
+    Output("navbar-collapse", "is_open"),
+    Input("navbar-toggler", "n_clicks"),
+    State("navbar-collapse", "is_open"),
+)
+def toggle_navbar_collapse(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+@APP.callback(
+    Output("hover_feature_label", "children"),
+    Input({"type": "borders_adm", "index": ALL}, "hover_feature")
+)
+def write_hover_adm_label(adm_loc):
+    location_description = "the map will return location name"
+    for i, adm in enumerate(adm_loc):
+        if adm is not None:
+            location_description = adm['geometry']['label']
+    return f'Mousing over {location_description}'
+
+
+@APP.callback(
+    Output("loc_marker", "position"),
+    Output("lat_input", "value"),
+    Output("lng_input", "value"),
+    Input("submit_coords","n_clicks"),
+    Input("map", "click_lat_lng"),
+    State("lat_input", "value"),
+    State("lng_input", "value")
+)
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+    if dash.ctx.triggered_id == None:
+        lat = rr_mrg["Y"][int(rr_mrg["Y"].size/2)].values
+        lng = rr_mrg["X"][int(rr_mrg["X"].size/2)].values
+    else:
+        if dash.ctx.triggered_id == "map":
+            lat = click_lat_lng[0]
+            lng = click_lat_lng[1]
+        else:
+            lat = latitude
+            lng = longitude
+        try:
+            nearest_grid = pingrid.sel_snap(rr_mrg, lat, lng)
+            lat = nearest_grid["Y"].values
+            lng = nearest_grid["X"].values
+        except KeyError:
+            lat = lat
+            lng = lng
+    return [lat, lng], lat, lng
+
+
+def crop_suitability(
+    rainfall_data,
+    min_wet_days,
+    wet_day_def,
+    tmax_data,
+    tmin_data,
+    lower_wet_threshold,
+    upper_wet_threshold,
+    max_temp,
+    min_temp,
+    temp_range,
+    target_season,
+):
+    seasonal_precip = rainfall_data.sel(
+        T=rainfall_data['T.season']==target_season
+    ).load()
+    seasonal_tmax = tmax_data.sel(T=tmax_data['T.season']==target_season).load()
+    seasonal_tmin = tmin_data.sel(T=tmin_data['T.season']==target_season).load()
+    sum_precip = seasonal_precip.groupby("T.year").sum("T")
+    avg_tmax = seasonal_tmax.groupby("T.year").mean("T")
+    avg_tmin = seasonal_tmin.groupby("T.year").mean("T")
+
+    #calculate average daily temperature range
+    avg_daily_temp_range = (
+        seasonal_tmax - seasonal_tmin
+    ).groupby("T.year").mean("T")
+    
+    total_wet_days = xr.where(
+        seasonal_precip >= float(wet_day_def),1,0
+    ).groupby("T.year").sum("T")
+    
+    #calculate total precip
+    total_precip_range = xr.where(
+        np.logical_and(
+            sum_precip <= float(upper_wet_threshold), 
+            sum_precip >= float(lower_wet_threshold)
+        ),1, 0)
```

Comment:
I think line 280-284 is equivalent to
```
total_precip_range = (
    (sum_precip <= float(upper_wet_threshold)) &&
    (sum_precip >= float(lower_wet_threshold))
)
```
which is a little easier to read, and yields a DataArray instead of a numpy array, which might or might not make it more convenient to use later. Similiarly below.

Reply:
I overall simplified the crop_suitability function for this and the previous comment. Have a look. I did have to multiply the variables by 1 to make sure they convert in numbers and don't stay boolean even though I don't really understand why.

Reply:
Yes, I see why you have to multiply by 1. If x and y are boolean arrays then the + in  `x + y` performs Boolean OR, even though `x.sum()` is addition, not Boolean OR. Arguably inconsistent behavior in numpy.